store/details.go

```diff
@@ -23,95 +23,96 @@ import (
+	Channel          puritan.String            `json:"channel,omitempty"`
+	DownloadSha3_384 puritan.SimpleString      `json:"download_sha3_384,omitempty"`
+	Summary          puritan.String            `json:"summary,omitempty"`
+	Description      puritan.Paragraph         `json:"description,omitempty"`
+	Deltas           []snapDeltaDetail         `json:"deltas,omitempty"`
+	DownloadSize     int64                     `json:"binary_filesize,omitempty"`
+	DownloadURL      puritan.String            `json:"download_url,omitempty"`
+	Epoch            snap.Epoch                `json:"epoch"`
+	IconURL          puritan.String            `json:"icon_url"`
+	LastUpdated      puritan.String            `json:"last_updated,omitempty"`
+	Name             puritan.SimpleString      `json:"package_name"`
+	Prices           puritan.OldPriceMap       `json:"prices,omitempty"`
 	// Note that the publisher is really the "display name" of the
 	// publisher
-	Publisher      string   `json:"publisher,omitempty"`
-	RatingsAverage float64  `json:"ratings_average,omitempty"`
-	Revision       int      `json:"revision"` // store revisions are ints starting at 1
-	ScreenshotURLs []string `json:"screenshot_urls,omitempty"`
-	SnapID         string   `json:"snap_id"`
-	SnapYAML       string   `json:"snap_yaml_raw"`
-	License        string   `json:"license,omitempty"`
-	Base           string   `json:"base,omitempty"`
+	Publisher      puritan.String       `json:"publisher,omitempty"`
+	RatingsAverage float64              `json:"ratings_average,omitempty"`
+	Revision       int                  `json:"revision"` // store revisions are ints starting at 1
+	ScreenshotURLs puritan.StringSlice  `json:"screenshot_urls,omitempty"`
+	SnapID         puritan.SimpleString `json:"snap_id"`
+	SnapYAML       puritan.Paragraph    `json:"snap_yaml_raw"`
+	License        puritan.String       `json:"license,omitempty"`
+	Base           puritan.SimpleString `json:"base,omitempty"`
 
 	// FIXME: the store should send "contact" here, once it does we
 	//        can remove support_url
-	SupportURL string `json:"support_url"`
-	Contact    string `json:"contact"`
+	SupportURL puritan.String `json:"support_url"`
+	Contact    puritan.String `json:"contact"`
 
-	Title   string    `json:"title"`
-	Type    snap.Type `json:"content,omitempty"`
-	Version string    `json:"version"`
+	Title   puritan.String `json:"title"`
+	Type    snap.Type      `json:"content,omitempty"`
+	Version puritan.String `json:"version"`
 
 	// TODO: have the store return a 'developer_username' for this
-	Developer   string `json:"origin"`
-	DeveloperID string `json:"developer_id"`
+	Developer   puritan.SimpleString `json:"origin"`
+	DeveloperID puritan.SimpleString `json:"developer_id"`
 
-	Private     bool   `json:"private"`
-	Confinement string `json:"confinement"`
+	Private     bool                 `json:"private"`
+	Confinement snap.ConfinementType `json:"confinement"`
```

Comment:
have we rechecked why it was a string?

Reply:
whatever it was, it's no longer needed (everything works fine without). Likely for transition.